microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/internal/openapi/java/MicroProfileGenerateOpenAPIOperation.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.openapi.java;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.TypeDeclaration;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+
+import com.redhat.microprofile.jdt.core.java.codeaction.IJavaCodeActionParticipant;
+import com.redhat.microprofile.jdt.core.java.codeaction.JavaCodeActionContext;
+import com.redhat.microprofile.jdt.core.java.corrections.proposal.ChangeCorrectionProposal;
+import com.redhat.microprofile.jdt.core.java.corrections.proposal.OpenAPIAnnotationProposal;
+import com.redhat.microprofile.jdt.internal.openapi.MicroProfileOpenAPIConstants;
+
+/**
+ * Generate OpenAPI annotations by the "Source" kind code action.
+ * 
+ * @author
+ *
+ */
+public class MicroProfileGenerateOpenAPIOperation implements IJavaCodeActionParticipant {
+
+	@Override
```

Comment:
Talked to Eric, he suggested to use a different issue to address that.

Reply:
IMHO I think it can be a good idea to check that there is the OpenAPI annotation in classpath, to provide this code action, no? What is your suggestion?

Reply:
You prefer me to add `isAdaptedForDiagnostics` to `IJavaCodeActionParticipant`, and also add the API to `ImplementHealthCheckQuickFix`, `InsertAnnotationMissingQuickFix`, `JavaCodeActionDefinition`? they all implement the interface, or I just add it to `MicroProfileGenerateOpenAPIOperation` only?

Reply:
well, maybe use the name like `isAdaptedForCodeActions` for the API

Reply:
if we add the API to the interface, then we need to add proper code to each implementation as well

Reply:
> well, maybe use the name like isAdaptedForCodeActions for the API

yes please.

> if we add the API to the interface, then we need to add proper check to each implementation as well

No the idea is that isAdaptedForCodeActions  return true by default. Please see the IJavaDiagnosticsParticipant.

Reply:
> Talked to Eric, he suggested to use a different issue to address that.

You mean that you prefer doing that in an another PR?

Reply:
Ok, i will add `isAdaptedForCodeAction` to `IJavaCodeActionParticipant` and it will return true by default, this will by pass for the other implementations (others won't do any check). In the  `MicroProfileGenerateOpenAPIOperation`, I will add the code to check the annotation and call the API before populating the code actions.

Reply:
Added it. 
BTW, i think we can consider to add the API call to a central place (e.g. JavaCodeActionDefinition or CodeActionHandler) for all.

Reply:
Indeed the isAdaptedForCodeAction should be managed by CodeActionHandler like diagnostics does. is there any chance that you do that in this PR? You can see how diagnostics does that.

Reply:
Done that.